pkg/deploy/generator/resources_disk_encryption.go

```diff
@@ -0,0 +1,119 @@
+package generator
+
+import (
+	mgmtcompute "github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2020-06-01/compute"
+	mgmtkeyvault "github.com/Azure/azure-sdk-for-go/services/keyvault/mgmt/2019-09-01/keyvault"
+	"github.com/Azure/go-autorest/autorest/to"
+
+	"github.com/Azure/ARO-RP/pkg/util/arm"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient"
+)
+
+func (g *generator) diskEncryptionKeyVault() *arm.Resource {
```

Comment:
This is very similar to above keyvault function. Is there way to write one generic function to return keyvault and inject rules? Can be smaller re-usable one (See how RBAC is being generated)